### PR TITLE
Add `lint-api` target and fix lint errors

### LINKED ISSUE #6153: `make lint` doesn't lint `api` directory
```
$ docker run --rm -v $(pwd):/go/src/github.com/gravitational/teleport -w /go/src/github.com/gravitational/teleport golangci/golangci-lint:v1.38.0 /bin/bash -c "cd api && golangci-lint run -c ../.golangci.yml ./..."
client/client.go:89:40: `guarentee` is a misspelling of `guarantee` (misspell)
// direct access to server values that guarentee a successful connection.
                                       ^
client/client.go:263:67: `propogate` is a misspelling of `propagate` (misspell)
                                        // Some credentials don't require these fields. If no errors propogate, then they need to provide these fields.
                                                                                                     ^
client/credentials_test.go:162:43: unnecessary conversion (unconvert)
        err = ioutil.WriteFile(tlsCasPath, []byte(tlsCACert), 0600)
                                                 ^
client/credentials_test.go:165:44: unnecessary conversion (unconvert)
        err = ioutil.WriteFile(sshCertPath, []byte(sshCert), 0600)
                                                  ^
client/credentials_test.go:168:43: unnecessary conversion (unconvert)
        err = ioutil.WriteFile(sshCasPath, []byte(sshCACert), 0600)
                                                 ^
client/credentials_test.go:156:40: unnecessary conversion (unconvert)
        err = ioutil.WriteFile(keyPath, []byte(keyPEM), 0600)
                                              ^
client/credentials_test.go:159:44: unnecessary conversion (unconvert)
        err = ioutil.WriteFile(tlsCertPath, []byte(tlsCert), 0600)
                                                  ^
types/access_request.go:231:27: func `(*AccessRequestV3).getThreshold` is unused (unused)
func (r *AccessRequestV3) getThreshold(tid uint32) (AccessReviewThreshold, error) {
                          ^
```

It seems that because `api` isn't technically part of the `teleport` package any more, `golangci-lint` won't lint it:

```
$ go list ./api/...
pattern ./api/...: main module (github.com/gravitational/teleport) does not contain package github.com/gravitational/teleport/api/client

$ docker run --rm -v $(pwd):/go/src/github.com/gravitational/teleport -w /go/src/github.com/gravitational/teleport golangci/golangci-lint:v1.38.0 /bin/bash -c "golangci-lint run -c .golangci.yml ./api/..."   
level=error msg="Running error: context loading failed: no go files to analyze"
```

We should probably add a separate `lint-api` make target which lints the `api` package separately using the same rules, and run that as part of `make lint`.

### PR DESCRIPTION
**Pull request by webvictim**
Fixes https://github.com/gravitational/teleport/issues/6153

- Adds a `lint-api` Makefile target which will run as part of the regular CI linting
- Fixes the current lint failures
- Changes the owner of `/go` to `ci:ci` to allow necessary dependencies to be downloaded

@Joerger I'm presuming that `getThreshold` is being added to support upcoming functionality, so added a `nolint` directive to avoid us needing to delete it.

### CODE REVIEW

**Comment by Joerger** [comment]
I think you meant to @fspmarshall about `getThreshold`.

File: api/types/access_request.go [hunk 1]
```diff
@@ -228,7 +228,7 @@ func (r *AccessRequestV3) GetOriginalRoles() []string {
 }
 
 // getThreshold is a helper for getting a threshold by its stored index.
-func (r *AccessRequestV3) getThreshold(tid uint32) (AccessReviewThreshold, error) {
+func (r *AccessRequestV3) getThreshold(tid uint32) (AccessReviewThreshold, error) { //nolint:unused
```
Comment (fspmarshall) [inline api/types/access_request.go:231]: actually this was just an oversight on my part (stopped being used in a last-minute refactor).  feel free to delete.

**Review by fspmarshall — APPROVED** [review]

**Review by Joerger — APPROVED** [review]

**Review by andrejtokarcik — APPROVED** [review]

**Review by awly — APPROVED** [review]

**Review by russjones — COMMENTED** [review]
Bot.

**Review by russjones — APPROVED** [review]
Bot.

**Pull request merged**